### ISSUE TITLE
Interface 'Options' incorrectly extends interface 'Partial<ProcessedOptions>'

### ISSUE THREAD

**Issue by jscheid** [jantimon/html-webpack-plugin#1552]
Hi, with `5.0.0-alpha.10` and typescript 4.0.5 I'm getting this:

```
node_modules/html-webpack-plugin/typings.d.ts:36:13 - error TS2430: Interface 'Options' incorrectly extends interface 'Partial<ProcessedOptions>'.
  Types of property 'filename' are incompatible.
    Type 'string | ((entryName: string) => string)' is not assignable to type 'string'.
      Type '(entryName: string) => string' is not assignable to type 'string'.

36   interface Options extends Partial<ProcessedOptions> {
               ~~~~~~~

```

**Comment by jantimon** [comment]
Oh that was changed in the latest version - I probably broke it... sorry for that - do you already have an idea what might causing the bug?

**Comment by jscheid** [comment]
No worries -- I'm not sure, but I guess if filename can be a function, then the signature in `ProcessedOptions` should be changed to reflect that. And vice versa, if it can only be a string, the signature in `Options` should be changed. Bottom line is that the two should match.

**Comment by jantimon** [comment]
Can you please try if `5.0.0-alpha.11` fixes this problem for you?

**Comment by jscheid** [comment]
I'm afraid it doesn't...

Like the error message says, the problem is that this:

https://github.com/jantimon/html-webpack-plugin/blob/2c57a1050b56c2ce96269e1c6342c0ee613a54f1/typings.d.ts#L37

is incompatible with this:

https://github.com/jantimon/html-webpack-plugin/blob/2c57a1050b56c2ce96269e1c6342c0ee613a54f1/typings.d.ts#L77

because the `((entryName: string) => string)` type alternative is missing.

So, a `ProcessedOption` cannot be coerced into an `Option` because it might contain a function for `filename` and that would violate the typing of `Option`.

**Comment by jantimon** [comment]
Fixed in 5.0.0-alpha.12:

https://github.com/jantimon/html-webpack-plugin/blob/f15f8e5c695781f4f6a18a68b156da6a3dc356f3/typings.d.ts#L29-L162
